src/main/java/uk/gov/register/indexer/IndexDriver.java

```diff
@@ -17,16 +17,17 @@ public IndexDriver(DataAccessLayer dataAccessLayer) {
     }
 
     public void indexEntry(Register register, Entry entry, IndexFunction indexFunction) {
-        Optional<Record> currentRecord = register.getRecord(entry.getKey());
+        Optional<Record> currentRecord = register.getDerivationRecord(entry.getKey(), indexFunction.getName());
```

Comment:
This doesn't look correct. It should always get the record. i.e. it should always be getting the current record for the `records` index no matter what index is being written.